api/tests/lib/seed_local_db.py

```diff
@@ -142,13 +161,28 @@ def _build_agencies(db_session: db.Session) -> None:
     # Create a static set of agencies, only if they don't already exist
     agencies = db_session.query(Agency).all()
     agency_codes = set([a.agency_code for a in agencies])
+    agency_dict = {a.agency_code: a for a in agencies}
```

Comment:
Wouldn't this approach require running this twice?

If you run with a fresh DB, this dict will be empty on the first run.

Maybe something like:

```py
agency_dict = {a.agency_code: a for a in agencies}

for agency_to_create in AGENCIES_TO_CREATE:
     ...

     new_agency = ...
     
     agency_dict[new_agency.agency_code] = new_agency

# a second pass
for agency_to_create in AGENCIES_TO_CREATE:
     ...
```

After typing that, it seems like this is literally just a separate version of what we'll implement in https://github.com/HHS/simpler-grants-gov/issues/2492 - it might just be easier to reuse that here once written. What do you think?

Reply:
Yeah, makes sense. What do you think about the latest change? It now works in 1 pass (last time it worked only because I had previously seeded). We could circle back and add the #2492 version later.

Reply:
Another way that might actually work without any code changes here.

In the agency definition above, do something like:
```py
AGENCIES_TO_CREATE = [
    {
       "agency_id": 1,
        "agency_code": "USAID",
        "agency_name": "Agency for International Development",
    },
    {
       "agency_id": 2,
        "agency_code": "ARPAH",
        "agency_name": "Advanced Research Projects Agency for Health",
    },
    {
       "agency_id": 3,
        "agency_code": "DOC",
        "agency_name": "Agency for International Development",
    },
    {
       "agency_id": 4,
        "agency_code": "DOC-EDA",
        "agency_name": "Agency for International Development",
        "top_level_agency_id": 3, # DOC
    },
]
```

Reply:
That makes it much easier. One loop and we can preserve the old logic.